lib/matplotlib/rcsetup.py

```diff
@@ -914,24 +914,37 @@ def _validate_linestyle(ls):
     A validator for all possible line styles, the named ones *and*
     the on-off ink sequences.
     """
-    # Named line style, like u'--' or u'solid'
-    if isinstance(ls, six.text_type):
-        return _validate_named_linestyle(ls)
-
-    # On-off ink (in points) sequence *of even length*.
+    # Look first for a valid named line style, like '--' or 'solid'
+    if isinstance(ls, six.string_types):
+        try:
+            return _validate_named_linestyle(ls)
+        except (UnicodeDecodeError, KeyError):
+            # On Python 2, string-like *ls*, like for example
+            # 'solid'.encode('utf-16'), may raise a unicode error.
+            raise ValueError("the linestyle string is not a valid string.")
+
+    if hasattr(ls, 'decode'):
```

Comment:
Huh, on Python 3.6, I get
```python
In [16]: hello = bytearray("Hello".encode('utf-16'))

In [17]: hasattr(hello, 'decode')
Out[17]: True
```
so this looks like bytearray instances *do* have a decode method, don't they?